pkg/haprovider/haprovider.go

```diff
@@ -364,13 +376,24 @@ func (r *HAProvider) CreateOrUpdateHAEndpoints(ctx context.Context, machine *clu
 		return err
 	}
 
+	// Get control plane endpoint ip family
+	adcForCluster, err := r.getADCForCluster(ctx, cluster)
```

Comment:
Do we need to check if the adc is v4 cluster is v6 here?

Reply:
We have added the check here: https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/blob/main/controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go#L361
I think we don't need to add this again?